src/main/java/duke/storage/Storage.java

```diff
@@ -80,8 +82,57 @@ private void read() throws FileLoadFailException {
         readBus();
         readTrain();
         readProfile();
+        readRecommendations();
         readEvent();
         readRoutes();
+        readItineraryTable();
+    }
+
+    /**
+     * Reads the itinerary hash map from storage.
+     *
+     * @throws FileLoadFailException If the file cannot be loaded.
+     */
+    public void readItineraryTable() throws FileLoadFailException {
+        itineraryTable = new HashMap<>();
+        try {
+            File itinerariesFile = new File(ITINERARIES_FILE_PATH);
+            Scanner scanner = new Scanner(itinerariesFile);
+            while (scanner.hasNextLine()) {
+                String name = scanner.nextLine();
+                LocalDateTime start = ParserTimeUtil.parseStringToDate(scanner.nextLine());
+                LocalDateTime end = ParserTimeUtil.parseStringToDate(scanner.nextLine());
+                Venue hotel = ParserStorageUtil.getVenueFromStorage(scanner.nextLine());
+                Itinerary itinerary = new Itinerary(start, end, hotel, name);
+                List<Agenda> agendaList = new ArrayList<>();
+                String fileLine = scanner.nextLine();
+                while (fileLine.split("\\|")[0].equals("Agenda ")) {
+                    List<Venue> venueList = new ArrayList<>();
+                    List<Todo> todoList;
+                    final int number2 = Integer.parseInt(fileLine.split("\\|")[1]);
+                    String newVenue = scanner.nextLine();
+                    while (newVenue.contains(" | ")) {
+                        venueList.add(ParserStorageUtil.getVenueFromStorage(newVenue));
+                        newVenue = scanner.nextLine();
+                    }
+                    todoList = ParserStorageUtil.getTodoListFromStorage(newVenue);
+                    Agenda agenda = new Agenda(todoList, venueList, number2);
+                    agendaList.add(agenda);
+                    if (scanner.hasNextLine()) {
+                        fileLine = scanner.nextLine();
+                    } else {
+                        break;
+                    }
+                }
+                itinerary.setTasks(agendaList);
+                itineraryTable.put(itinerary.getName(), itinerary);
+            }
+            scanner.close();
+        } catch (FileNotFoundException | ParseException e) {
+            itineraryTable = new HashMap<>();
```

Comment:
This `itineraryTable = new HashMap<>();` could be moved to the constructor

Reply:
Done